src/LinuxTracing/StackSlice.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef LINUX_TRACING_STACK_SLICE_H_
+#define LINUX_TRACING_STACK_SLICE_H_
+
+#include <absl/types/span.h>
+
+#include <memory>
+
+namespace orbit_linux_tracing {
+
+// This class holds a copy of some stack data from collected from the target process.
+// The class must have a single owner, to avoid unnecessary copies. Instead of passing pointers or
+// references of this class around, `StackSliceView` should be used instead.
+class StackSlice {
+ public:
+  StackSlice(uint64_t start_address, uint64_t size, std::unique_ptr<char[]>&& data)
+      : start_address_{start_address}, size_{size}, data_{std::move(data)} {}
+
+  [[nodiscard]] uint64_t GetStartAddress() const { return start_address_; }
+  [[nodiscard]] uint64_t GetSize() const { return size_; }
```

Comment:
Super Nit: For consistency reason I would also introduce a `GetEndAddress` function here.